graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -299,12 +245,12 @@ def _compile_property_ast(schema, current_schema_type, ast, location, context, l
         validate_safe_string(output_name)
 
         graphql_type = strip_non_null_from_type(current_schema_type)
-        if is_in_fold:
+        if is_in_fold_scope(context):
```

Comment:
just pointing out that this is _slightly_ different from the original logic

Reply:
Good catch. I did this on purpose since this is technically more correct, the original code was a bit of a premature optimization that I was worried would come back to bite at some point in the future.

Reply:
yah, in general this is all great -- seems like a good amount of abstraction, but not too much abstraction.  Just don't like, make a huge database request to check `is_in_fold_scope` :D

Reply:
That'd be ironic, checking the database 100 times so that it can create the perfect query that avoids checking the database 100 times... :)